src/app/pages/login/login.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { AuthService } from '../../services/auth.service';
+import { Router } from '@angular/router';
+import { ToastrService } from 'ngx-toastr';
+import { Store } from '@ngrx/store';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrl: './login.component.css'
+})
+export class LoginComponent implements OnInit{
+
+  loginForm!: FormGroup;
+  currentUser: any ;
+  submitInProgress = false;
+  error = "";
+
+  constructor(private fb: FormBuilder,
+              private authService: AuthService,
```

Comment:
private readonly authService: AuthService

Make the services readonly to prevent to modify